src/TensorValues/Operations.jl

```diff
@@ -222,6 +259,68 @@ end
 
 const ⊙ = inner
 
+###############################################################
+# Double Contractions w/ products
+###############################################################
+
+# a_i = b_ijk*c_jk
+@generated function double_contraction(a::A, b::B) where {A<:MultiValue{Tuple{D1,D2,D3}},B<:MultiValue{Tuple{D2,D3}}} where {D1,D2,D3}
+  ss = String[]
+  for i in 1:D1
+    s = join([ "a[$i,$j,$k]*b[$j,$k]+" for j in 1:D2 for k in 1:D3])
+    push!(ss,s[1:(end-1)]*", ")
+  end
+  str = join(ss)
+  Meta.parse("VectorValue{$D1}(($str))")
+end
+
+# a_ijpm = b_ijkl*c_klpm
+@generated function double_contraction(a::A, b::B) where {A<:SymFourthOrderTensorValue{D},B<:SymFourthOrderTensorValue{D}} where D
+
+  Sym4TensorIndexing = [1111, 1121, 1131, 1122, 1132, 1133, 2111, 2121, 2131, 2122, 2132, 2133,
```

Comment:
This is particular for 3D. Can you implement it in general?
If for performance reasons, you want to keep the 3D specialization then use the signature 

```julia
function double_contraction(a::A, b::B) where {A<:SymFourthOrderTensorValue{3},B<:SymFourthOrderTensorValue{3}}
```

Reply:
Okay, perhaps `⋅²`. I will keep the 3D version since this should be fairly quick. I will have to have a think about how to do this generally.

Reply:
This might do the trick:
```
@generated function double_contraction(a::SymFourthOrderTensorValue{D}, b::SymFourthOrderTensorValue{D}) where D
  str = ""
  for j in 1:D
    for i in j:D
      for m in 1:D
        for p in m:D
          s = ""
          for k in 1:D
            for l in 1:D
              s *= " a[$i,$j,$k,$l]*b[$k,$l,$p,$m] +"
            end
          end
          str *= s[1:(end-1)]*", "
        end
      end
    end
  end
  Meta.parse("SymFourthOrderTensorValue{D}($str)")
end
```

Need to test.

Reply:
Yes, keep ⋅² so we use the same notation for all >1 contractions.